### DISCUSSION TITLE
what is the difference between v3 and v4?

### DISCUSSION THREAD

**Discussion by lp2jx0309** [dropwizard/dropwizard#5694] [Q&A]

**Comment by joschi** [comment]
@lp2jx0309 Dropwizard 3.x will still be using the Java EE namespace (`javax.*`) while Dropwizard 4.x will be using the Jakarta EE namespace (`jakarta.*`).

The baseline for both releases will be Java 11.

**Reply by mattnelson** [reply]
> The baseline for both releases will be Java 11.

Has it been discussed if 4.x should baseline 17?

**Reply by erwinw** [reply]
> > The baseline for both releases will be Java 11.
> 
> Has it been discussed if 4.x should baseline 17?

Great question, @mattnelson . Any input (🙏🏼), @joschi?

**Reply by joschi** [reply]
@erwinw @mattnelson What value would a _baseline_ of Java 17 in Dropwizard 4.x provide to the users of Dropwizard?

You can already write Dropwizard 2.1.x, 3.x, and 4.x applications in Java 17 if you want.

**Reply by mattnelson** [reply]
> You can already write Dropwizard 2.1.x, 3.x, and 4.x applications in Java 17 if you want.

Full disclosure, I am already doing this.

> What value would a baseline of Java 17 in Dropwizard 4.x provide to the users of Dropwizard?

I don't have a strong argument for/against 17, but I wasn't sure if it has been previously discussed.
These items are kinda weak arguments, but it is what I can think of right now.

* It would be a hedge against requests to cut a 5.x if any of the libraries that dropwizard depends on uplift to 17. Are there any dep updates that 3.x/4.x have already not been able to take because of this?
* The perception of not being "modern" compared to other options that baseline on 17.

> Dropwizard 3.x will still be using the Java EE namespace (javax.*) while Dropwizard 4.x will be using the Jakarta EE namespace (jakarta.*).

Projecting a bit future out. Is this how you see the progression of even future dropwizard version?

|  dropwizard version | EE version | EE namespace | java version |
|---|---|---|---|
|  2.1.x | JavaEE8  | javax | 8  |
|  3.x | JakartaEE8  | javax | 11  |
|  4.x | JakartaEE9  | jakarta | 11  |
|  5.x | JakartaEE10  | jakarta | 11/17? |
|  6.x | JakartaEE11  | jakarta | 17 |
